### PR TITLE
Makefile: Setup gitlab on minikube using helm3

### PR DESCRIPTION
**Pull request by guy9050**
There are some extra steps needed because it is install on minikube.
All info is here https://docs.gitlab.com/charts/development/minikube/.

Instructions for the PR:
After running `make up`,wait till all the pods are up and running, before that you won't be able to access the private gitlab. Sometimes it takes 15-20 mins.
Once all the pods are running (`kubectl get pods -w`)
run `minikube ip`
open this in your browser: `https://gitlab.[minikube ip].nip.io/`

Get the password to login: `kubectl get secret gitlab-gitlab-initial-root-password -ojsonpath='{.data.password}' | base64 --decode ; echo`

### CODE REVIEW

**Review by lmilbaum — CHANGES_REQUESTED** [review]
Can you please fix the broken CI?

File: Makefile [hunk 1]
```diff
@@ -2,10 +2,15 @@
...
+	minikube addons enable ingress
+	minikube addons enable dashboard
 	kubectl create namespace gitlab
 	helm repo add gitlab https://charts.gitlab.io
 	helm repo update
-	helm install gitlab gitlab/gitlab --namespace gitlab --values values.yaml --version v7.1.0
-
+	helm upgrade --install gitlab gitlab/gitlab \
+	--timeout 600s \
+	--set global.hosts.domain=$$(minikube ip).nip.io \
```
Comment (lmilbaum) [inline Makefile:12]: Can we use our own values file instead?

File: Makefile [hunk 2]
```diff
@@ -2,10 +2,15 @@
 
 up:
 	minikube start
+	minikube addons enable ingress
+	minikube addons enable dashboard
 	kubectl create namespace gitlab
 	helm repo add gitlab https://charts.gitlab.io
 	helm repo update
-	helm install gitlab gitlab/gitlab --namespace gitlab --values values.yaml --version v7.1.0
```
Comment (lmilbaum) [inline Makefile:8]: When we should use the install command and when to use the upgrade command?

Comment (lmilbaum) [inline Makefile:8]: The pinned version is important for reproducible build.  Is there a reason to drop this and use the latest version every time it is used?

File: Makefile
Comment (lmilbaum) [inline Makefile:1]: I'd suggest to update the DEV environment setup in README

**Review by lmilbaum — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (guy9050) [reply Makefile:12]: We can, but I'll need to modify it

File: Makefile (hunk 2 above)
Comment (guy9050) [reply Makefile:8]: We can use both, but according to the best practices in this [article](https://codersociety.com/blog/articles/helm-best-practices) - rule #13 it's better to use upgrade with `--install` flag `Always use the helm upgrade --install command. It installs the charts if they are not already installed. If they are already installed, it upgrades them.`

File: Makefile
Comment (guy9050) [reply Makefile:1]: Done

File: Makefile (hunk 2 above)
Comment (guy9050) [reply Makefile:8]: Just so we're not stuck in some old version:) I brought it back

File: Makefile (hunk 1 above)
Comment (guy9050) [reply Makefile:14]: so now we're using our values file

**Comment by guy9050** [comment]
> Can you please fix the broken CI?

I fixed the previous issue, but the deployment requires more than 2 CPU, and the runner is limited to 2, so it's failing on this now:
`X Exiting due to RSRC_INSUFFICIENT_CORES: Requested cpu count 4 is greater than the available cpus of 2`

File: Makefile (hunk 2 above)
Comment (lmilbaum) [reply Makefile:8]: Thanks for sharing this info.   Not sure if the upgrade process is aligned with the reproducible build practice. Every external dependency is supposed to be pinned versioned.

**Comment by lmilbaum** [comment]
> > Can you please fix the broken CI?
> 
> I fixed the previous issue, but the deployment requires more than 2 CPU, and the runner is limited to 2, so it's failing on this now: `X Exiting due to RSRC_INSUFFICIENT_CORES: Requested cpu count 4 is greater than the available cpus of 2`

Can you check if a bigger runner can be used?

**Comment by lmilbaum** [comment]
The instructions are not working on my laptop. Can we hop on a call sometime to trouble shoot?

**Comment by guy9050** [comment]
> The instructions are not working on my laptop. Can we hop on a call sometime to trouble shoot?

Sure. I'll set up a mtg for Sunday

**Comment by guy9050** [comment]
> > > Can you please fix the broken CI?
> > 
> > 
> > I fixed the previous issue, but the deployment requires more than 2 CPU, and the runner is limited to 2, so it's failing on this now: `X Exiting due to RSRC_INSUFFICIENT_CORES: Requested cpu count 4 is greater than the available cpus of 2`
> 
> Can you check if a bigger runner can be used?

Yes, we could set up a larger runner on this page - https://github.com/organizations/platform-engineering-org/settings/actions/runners, I tried to do it, but unfortunately it requires a payment. "Only available on paid Team or enterprise". I will try to find a workaround.

**Pull request merged**